src/workloads/query/ProjectParse.yml

```diff
@@ -0,0 +1,91 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  Test $project parsing performance for small and large number of fields.
+
+GlobalDefaults:
+  dbname: &db test
+  coll: &coll Collection0
+  nop: &Nop {Nop: true}
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10
+    BatchSize: 10
+    Document:
+      m: {^Inc: {start: 0}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - *Nop
+
+- Name: Queries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 10000
+    Database: *db
+    Operations:
+    - OperationMetricsName: ProjectParseSmall
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *coll
+        pipeline: [
+          {$project : {^Object: {
+              withNEntries: 10,
+              havingKeys: {^RandomString: {length: 10}},
+              andValues: 1,
+              duplicatedKeys: skip
+            }}}
+        ]
+        cursor: {}
+  - Repeat: 50
```

Comment:
Why 50? Would you prefer to run it for  a set period of time instead?

Reply:
Ah yea 50 was sort of arbitrarily chosen, I think setting a time would be better. I'll change this